src/display/pattern_helper.js

```diff
@@ -87,8 +87,8 @@ class RadialAxialShadingPattern extends BaseShadingPattern {
 
   getPattern(ctx, owner, inverse, shadingFill = false) {
     let pattern;
-    if (this._patternCache) {
-      pattern = this._patternCache;
+    if (!shadingFill && this.cachedCanvasPatterns.has(this)) {
+      pattern = this.cachedCanvasPatterns.get(this);
     } else {
       if (!shadingFill) {
         const tmpCanvas = owner.cachedCanvases.getCanvas(
```

Comment:
Nit: Unless I'm completely misreading this code, it'd feel slightly easier to reason about the conditions with the following small change:
```suggestion
    if (!shadingFill) {
      if (this.cachedCanvasPatterns.has(this)) {
        pattern = this.cachedCanvasPatterns.get(this);
      } else {
        const tmpCanvas = owner.cachedCanvases.getCanvas(
```